src/vm/i386/asmhelpers.S

```diff
@@ -1162,20 +1162,30 @@ NESTED_END ResolveWorkerChainLookupAsmStub, _TEXT
 NESTED_ENTRY BackPatchWorkerAsmStub, _TEXT, NoHandler
     PROLOG_BEG
     PROLOG_PUSH eax // it may contain siteAddrForRegisterIndirect
+    PROLOG_PUSH ebx
     PROLOG_PUSH ecx
     PROLOG_PUSH edx
     PROLOG_END
 
-    sub     esp, 4     //  for 16 bytes align
+    // Align Stack Frame
+    mov     ebx, esp
+    and     esp, -16
+
+    // Call BackPatchWorkerStaticStub
+    sub     esp, 8     //  16-bytes align padding
```

Comment:
This stub is used by virtual calls dispatch. It is called by ResolveStub, see https://github.com/dotnet/coreclr/blob/master/src/vm/i386/virtualcallstubcpu.hpp#L250-L324.
So it seems that if there is an alignment issue, it would stem from the managed caller.
Could you please break in this function when the alignment is incorrect and see what is the stack trace and also disass the managed function that called it?